File: Task_1/count_ws/src/SubCount_pkg/src/sub.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"

// callback function to handle messgae
void callback(const std_msgs::Int32::ConstPtr& msg)  // ConstPtr: smart pointer that provides automatic memory management for the message.
{
    ROS_INFO("count = %d", msg->data); // print msg on screen | -> point to class
}

int main(int argc, char** argv)
{   
    // create node
    ros::init(argc, argv, "Subscriber");
    ros::NodeHandle nh;

    // create subscriber object
    ros::Subscriber sub;
    // subscribe message from counter topic
    sub = nh.subscribe("counter", 1, callback);
    ros::spin();

    return 0;
}

File: Task_1/count_ws/src/SubCount_pkg/scripts/sub.py
#!/usr/bin/python3

import rospy
from std_msgs.msg import Int32

def listener():
    # create node
    rospy.init_node("Listener")
    # subscribe count_topic
    rospy.Subscriber("count_topic", Int32, callback)
    rospy.spin() # processes incoming messages until the node is shut down.

def callback(count):   # callback function to handle msg
    rospy.loginfo("count = %d", count.data)



if __name__ == "__main__":
    listener()